### ISSUE TITLE
Unable to build with version 1.14.x

### ISSUE THREAD

**Issue by AndersHogqvist** [microsoft/vscode-cmake-tools#3123]
### Brief Issue Summary

Everything works just fine using version 1.13.45, but when switching to 1.14.x I'm no longer able to build. After configuration I get the following error:

`[rollbar] Unhandled exception: Unhandled Promise rejection: cleanRebuild TypeError: inputs is not iterable {}`

### CMake Tools Diagnostics

```shell
{
  "os": "linux",
  "vscodeVersion": "1.77.3",
  "cmtVersion": "1.14.30",
  "configurations": [
    {
      "folder": "/workspaces/xxxxxx",
      "cmakeVersion": "3.22.5",
      "configured": true,
      "generator": "Ninja",
      "usesPresets": false,
      "compilers": {
        "C": "/usr/bin/gcc",
        "CXX": "/usr/bin/g++"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": false
    }
  ]
}
```

### Debug Log

```shell
[main] Building folder: xxxxxxx clean
[main] Saving open files before configure/build
[main] Configuring project: xxxxxxx 
[main] Saving open files before configure/build
[driver] Start configure 
[driver] Running pre-configure checks and steps
[driver] Using compilers in native for configure
[cmakefileapi-driver] Configuring using kit
[cmakefileapi-driver] Invoking CMake /usr/bin/cmake with arguments ["--no-warn-unused-cli","-DXXXXXXXXX_ENABLE_DOCS:STRING=YES","-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE","-DCMAKE_BUILD_TYPE:STRING=Debug","-DCMAKE_C_COMPILER:FILEPATH=/usr/bin/gcc","-DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/g++","-S/workspaces/xxxxxxx","-B/workspaces/xxxxxxx/build/native","-G","Ninja"]
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DXXXXXXXXX_ENABLE_DOCS:STRING=YES -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/gcc -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/g++ -S/workspaces/xxxxxxx -B/workspaces/xxxxxxx/build/native -G Ninja
[proc]   with environment: {"HOSTNAME":"6161b6e6573f","HOME":"/home/vscode","PATH":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710/bin/remote-cli:/home/vscode/.local/bin:/home/vscode/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/vscode/.dotnet/tools","VSCODE_AGENT_FOLDER":"/home/vscode/.vscode-server","SHELL":"/bin/bash","PWD":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710","VSCODE_HANDLES_SIGPIPE":"true","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_NLS_CONFIG":"{\"locale\":\"en\",\"osLocale\":\"en\",\"availableLanguages\":{}}","REMOTE_CONTAINERS_IPC":"/tmp/vscode-remote-containers-ipc-ccac41ac-3844-4504-a48d-3dac4d5baf9d.sock","LS_COLORS":"","LESSCLOSE":"/usr/bin/lesspipe %s %s","REMOTE_CONTAINERS_SOCKETS":"[\"/tmp/vscode-ssh-auth-ccac41ac-3844-4504-a48d-3dac4d5baf9d.sock\",\"/tmp/.X11-unix/X75\",\"/home/vscode/.gnupg/S.gpg-agent\"]","LESSOPEN":"| /usr/bin/lesspipe %s","SHLVL":"1","_":"/usr/bin/cat","SSH_AUTH_SOCK":"/tmp/vscode-ssh-auth-ccac41ac-3844-4504-a48d-3dac4d5baf9d.sock","DISPLAY":":75","REMOTE_CONTAINERS_DISPLAY_SOCK":"/tmp/.X11-unix/X75","XDG_RUNTIME_DIR":"/tmp/user/1000","WAYLAND_DISPLAY":"vscode-wayland-6d3ca7c47eb744e6cb620b50d97c1e3e463d9ae2.sock","REMOTE_CONTAINERS":"true","BROWSER":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710/bin/helpers/browser.sh","VSCODE_CWD":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710","ELECTRON_RUN_AS_NODE":"1","VSCODE_IPC_HOOK_CLI":"/tmp/user/1000/vscode-ipc-363f64f9-8ca5-4278-8fdc-05d03e165b9e.sock","VSCODE_L10N_BUNDLE_LOCATION":"","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"1","BUILD_DIR":"build"}
[cmake] Not searching for unused variables given on the command line.
[cmake] -- The C compiler identification is GNU 9.4.0
[cmake] -- The CXX compiler identification is GNU 9.4.0
[cmake] -- Detecting C compiler ABI info
[cmake] -- Detecting C compiler ABI info - done
[cmake] -- Check for working C compiler: /usr/bin/gcc - skipped
[cmake] -- Detecting C compile features
[cmake] -- Detecting C compile features - done
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /usr/bin/g++ - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- Found Git: /usr/bin/git (found version "2.25.1") 
[cmake] -- Found GetVersion: /home/vscode/.local/bin/getversion (found version "GetVersion 6.2.6") 
[cmake] Set Prefix UZONE_7654321
[cmake] Set value UZONE_7654321 for getversion.prefix
[cmake] GetVersion 6.2.6
[cmake] ---
[cmake] Current dir /workspaces/xxxxxxx
[cmake] Buildnr loaded from git config
[cmake] Prefix loaded from git config
[cmake] Prefix UZONE_7654321
[cmake] Buildnr 1
[cmake] Updatebuildnr None
[cmake] Clearcache True
[cmake] Showvariable None
[cmake] Local Development
[cmake] Current branch is main
[cmake] Current commit is a3d48d6abde9602fadfb29e06fe4828f0c879fdc (a3d48d6)
[cmake] Git folder is .git
[cmake] Cache folder is .git/getversion_cache
[cmake] Clearing cache files from .git/getversion_cache
[cmake] -- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found version "1.71.0")  
[cmake] -- Looking for pthread.h
[cmake] -- Looking for pthread.h - found
[cmake] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[cmake] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
[cmake] -- Looking for pthread_create in pthreads
[cmake] -- Looking for pthread_create in pthreads - not found
[cmake] -- Looking for pthread_create in pthread
[cmake] -- Looking for pthread_create in pthread - found
[cmake] -- Found Threads: TRUE  
[cmake] -- Check if compiler accepts -pthread
[cmake] -- Check if compiler accepts -pthread - yes
[cmake] -- Looking for ndc8_error_string in ndc8lib
[cmake] -- Looking for ndc8_error_string in ndc8lib - not found
[cmake] -- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
[cmake] -- Checking for module 'eigen3'
[cmake] --   Found eigen3, version 3.3.7
[cmake] -- Checking for module 'flann'
[cmake] --   Found flann, version 1.9.1
[cmake] -- Found FLANN: /usr/lib/x86_64-linux-gnu/libflann_cpp_s.a  
[cmake] -- FLANN found (include: /usr/include, lib: /usr/lib/x86_64-linux-gnu/libflann_cpp_s.a)
[cmake] -- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found version "1.71.0") found components: iostreams 
[cmake] -- Checking for module 'eigen3'
[cmake] --   Found eigen3, version 3.3.7
[cmake] -- Found Eigen: /usr/include/eigen3 (Required is at least version "3.3") 
[cmake] -- Eigen found (include: /usr/include/eigen3, version: 3.3.7)
[cmake] -- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found suitable version "1.71.0", minimum required is "1.65.0") found components: system filesystem date_time iostreams serialization 
[cmake] -- Eigen found (include: /usr/include/eigen3, version: 3.3.7)
[cmake] -- looking for PCL_COMMON
[cmake] -- Found PCL_COMMON: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_common.so  
[cmake] -- looking for PCL_OCTREE
[cmake] -- Found PCL_OCTREE: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_octree.so  
[cmake] -- looking for PCL_IO
[cmake] -- Found PCL_IO: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_io.so  
[cmake] -- looking for PCL_KDTREE
[cmake] -- Found PCL_KDTREE: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_kdtree.so  
[cmake] -- looking for PCL_SEARCH
[cmake] -- Found PCL_SEARCH: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_search.so  
[cmake] -- looking for PCL_SAMPLE_CONSENSUS
[cmake] -- Found PCL_SAMPLE_CONSENSUS: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_sample_consensus.so  
[cmake] -- looking for PCL_FILTERS
[cmake] -- Found PCL_FILTERS: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_filters.so  
[cmake] -- looking for PCL_ML
[cmake] -- Found PCL_ML: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_ml.so  
[cmake] -- looking for PCL_GEOMETRY
[cmake] -- Found PCL_GEOMETRY: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/include/pcl-1.13  
[cmake] -- looking for PCL_2D
[cmake] -- Found PCL_2D: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/include/pcl-1.13  
[cmake] -- looking for PCL_FEATURES
[cmake] -- Found PCL_FEATURES: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_features.so  
[cmake] -- looking for PCL_SEGMENTATION
[cmake] -- Found PCL_SEGMENTATION: /workspaces/xxxxxxx/src/LaserScanner/third-party/pcl/x86/lib/libpcl_segmentation.so  
[cmake] -- Found PCL: pcl_common;pcl_octree;pcl_io;pcl_kdtree;pcl_search;pcl_sample_consensus;pcl_filters;pcl_ml;pcl_features;pcl_segmentation;Boost::system;Boost::filesystem;Boost::date_time;Boost::iostreams;Boost::serialization;FLANN::FLANN  
[cmake] -- Checking for one of the modules 'gtest_main'
[cmake] -- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found components: doxygen dot 
[cmake] Doxygen build started
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: /workspaces/xxxxxxx/build/native
[cmakefileapi-parser] Read reply folder: /workspaces/xxxxxxx/build/native/.cmake/api/v1/reply
[cmakefileapi-parser] Found index files: ["cache-v2-6e0a565f4707b0673d30.json","cmakeFiles-v1-fa3a04c65c4d622e0642.json","codemodel-v2-7f88f24f4d9d4fe67b7c.json","directory-.-Debug-f5ebdc15457944623624.json","directory-Ndc8-Debug-43329f0e66a8aa6718ef.json","directory-Ndc8.Parameter-Debug-2687a7be1e19232c02a4.json","directory-docs-Debug-c463161ff6ebad611835.json","directory-src-Debug-b6cf2213fc5053b67c22.json","directory-src.LaserScanner-Debug-777930562b888e8db903.json","directory-src.LaserScanner.test-Debug-be16d2106d865d7f92d9.json","directory-src.test-Debug-3607701395add829ebf1.json","index-2023-04-21T06-33-24-0794.json","target-Continuous-Debug-c38766585cd35c558d93.json","target-ContinuousBuild-Debug-af6e8bed105297a9dc48.json","target-ContinuousConfigure-Debug-77a9c61d6b29e8201d05.json","target-ContinuousCoverage-Debug-41d47a679bc20eaaab29.json","target-ContinuousMemCheck-Debug-e5e2909d0ff2b1276e5c.json","target-ContinuousStart-Debug-de5512f9e7430be4d52f.json","target-ContinuousSubmit-Debug-d2b568fcd33d6bab0a60.json","target-ContinuousTest-Debug-33d7aee7cd6d590e387d.json","target-ContinuousUpdate-Debug-5ba7a21decd211c8e0a3.json","target-Experimental-Debug-1b2521dd338a7c88b479.json","target-ExperimentalBuild-Debug-df55c57996889b2f8ee2.json","target-ExperimentalConfigure-Debug-50e027e96cf3ecc4ac0c.json","target-ExperimentalCoverage-Debug-6e8fe3983dd72bed038a.json","target-ExperimentalMemCheck-Debug-56d63ebf39dada5217a6.json","target-ExperimentalStart-Debug-b807cf905886c602a630.json","target-ExperimentalSubmit-Debug-487eba5b10d2a42668bc.json","target-ExperimentalTest-Debug-51c70e6ea971089258d9.json","target-ExperimentalUpdate-Debug-17e7d97ea99324d94b15.json","target-Nightly-Debug-02dd6253a806c7b4921f.json","target-NightlyBuild-Debug-b90a246f1467d06276b1.json","target-NightlyConfigure-Debug-b11dca70f55dc20d4455.json","target-NightlyCoverage-Debug-ce4b96944ffe5b035829.json","target-NightlyMemCheck-Debug-c5e1e5fe032c03433c5e.json","target-NightlyMemoryCheck-Debug-7d77d6cd5d22eb2e2974.json","target-NightlyStart-Debug-5fdc4b6d0e4b07142b2c.json","target-NightlySubmit-Debug-3289d02ff4ab3279210d.json","target-NightlyTest-Debug-b16778a2eb1a15bd02ed.json","target-NightlyUpdate-Debug-89959c7a76a9003a42fc.json","target-docs-xxxxxxxx-html_zip-Debug-bba875f687bb0116730a.json","target-doxydoc-Debug-cb0e1dd8198d3cb35b9a.json","target-doxygen_doc-Debug-c43e59f5560a2cf6a3ec.json","target-laserscanner-Debug-3591bf9ce61cf09b363b.json","target-ndc8_parameter-Debug-6e53908ec7feeca701eb.json","target-test_bb_handler-Debug-9f74d551ebf92cff671d.json","target-test_laserscanner-Debug-8a64c4684efe5f696f94.json","toolchains-v1-2781238ec0fa0b9996f9.json"]
[extension] Not updating the configuration provider because "C_Cpp.intelliSenseEngine" is set to "Disabled"
[main] Copying compile_commands.json from /workspaces/xxxxxxx/build/native/compile_commands.json to /workspaces/xxxxxxx/compile_commands.json
[build] Starting build
[driver] Start build clean
[proc] Executing command: /usr/bin/cmake --build /workspaces/xxxxxxx/build/native --config Debug --target clean --
[proc]   with environment: {"HOSTNAME":"6161b6e6573f","HOME":"/home/vscode","PATH":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710/bin/remote-cli:/home/vscode/.local/bin:/home/vscode/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/vscode/.dotnet/tools","VSCODE_AGENT_FOLDER":"/home/vscode/.vscode-server","SHELL":"/bin/bash","PWD":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710","VSCODE_HANDLES_SIGPIPE":"true","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_NLS_CONFIG":"{\"locale\":\"en\",\"osLocale\":\"en\",\"availableLanguages\":{}}","REMOTE_CONTAINERS_IPC":"/tmp/vscode-remote-containers-ipc-ccac41ac-3844-4504-a48d-3dac4d5baf9d.sock","LS_COLORS":"","LESSCLOSE":"/usr/bin/lesspipe %s %s","REMOTE_CONTAINERS_SOCKETS":"[\"/tmp/vscode-ssh-auth-ccac41ac-3844-4504-a48d-3dac4d5baf9d.sock\",\"/tmp/.X11-unix/X75\",\"/home/vscode/.gnupg/S.gpg-agent\"]","LESSOPEN":"| /usr/bin/lesspipe %s","SHLVL":"1","_":"/usr/bin/cat","SSH_AUTH_SOCK":"/tmp/vscode-ssh-auth-ccac41ac-3844-4504-a48d-3dac4d5baf9d.sock","DISPLAY":":75","REMOTE_CONTAINERS_DISPLAY_SOCK":"/tmp/.X11-unix/X75","XDG_RUNTIME_DIR":"/tmp/user/1000","WAYLAND_DISPLAY":"vscode-wayland-6d3ca7c47eb744e6cb620b50d97c1e3e463d9ae2.sock","REMOTE_CONTAINERS":"true","BROWSER":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710/bin/helpers/browser.sh","VSCODE_CWD":"/vscode/vscode-server/bin/linux-x64/704ed70d4fd1c6bd6342c436f1ede30d1cff4710","ELECTRON_RUN_AS_NODE":"1","VSCODE_IPC_HOOK_CLI":"/tmp/user/1000/vscode-ipc-363f64f9-8ca5-4278-8fdc-05d03e165b9e.sock","VSCODE_L10N_BUNDLE_LOCATION":"","APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL":"1","BUILD_DIR":"build","NINJA_STATUS":"[%s/%t %p :: %e] "}
[build] [1/1 100% :: 0.003] Cleaning all built files...
[build] Cleaning... 0 files.
[driver] Build completed: 00:00:00.028
[cmakefileapi-parser] Read reply folder: /workspaces/xxxxxxx/build/native/.cmake/api/v1/reply
[cmakefileapi-parser] Found index files: ["cache-v2-6e0a565f4707b0673d30.json","cmakeFiles-v1-fa3a04c65c4d622e0642.json","codemodel-v2-7f88f24f4d9d4fe67b7c.json","directory-.-Debug-f5ebdc15457944623624.json","directory-Ndc8-Debug-43329f0e66a8aa6718ef.json","directory-Ndc8.Parameter-Debug-2687a7be1e19232c02a4.json","directory-docs-Debug-c463161ff6ebad611835.json","directory-src-Debug-b6cf2213fc5053b67c22.json","directory-src.LaserScanner-Debug-777930562b888e8db903.json","directory-src.LaserScanner.test-Debug-be16d2106d865d7f92d9.json","directory-src.test-Debug-3607701395add829ebf1.json","index-2023-04-21T06-33-24-0794.json","target-Continuous-Debug-c38766585cd35c558d93.json","target-ContinuousBuild-Debug-af6e8bed105297a9dc48.json","target-ContinuousConfigure-Debug-77a9c61d6b29e8201d05.json","target-ContinuousCoverage-Debug-41d47a679bc20eaaab29.json","target-ContinuousMemCheck-Debug-e5e2909d0ff2b1276e5c.json","target-ContinuousStart-Debug-de5512f9e7430be4d52f.json","target-ContinuousSubmit-Debug-d2b568fcd33d6bab0a60.json","target-ContinuousTest-Debug-33d7aee7cd6d590e387d.json","target-ContinuousUpdate-Debug-5ba7a21decd211c8e0a3.json","target-Experimental-Debug-1b2521dd338a7c88b479.json","target-ExperimentalBuild-Debug-df55c57996889b2f8ee2.json","target-ExperimentalConfigure-Debug-50e027e96cf3ecc4ac0c.json","target-ExperimentalCoverage-Debug-6e8fe3983dd72bed038a.json","target-ExperimentalMemCheck-Debug-56d63ebf39dada5217a6.json","target-ExperimentalStart-Debug-b807cf905886c602a630.json","target-ExperimentalSubmit-Debug-487eba5b10d2a42668bc.json","target-ExperimentalTest-Debug-51c70e6ea971089258d9.json","target-ExperimentalUpdate-Debug-17e7d97ea99324d94b15.json","target-Nightly-Debug-02dd6253a806c7b4921f.json","target-NightlyBuild-Debug-b90a246f1467d06276b1.json","target-NightlyConfigure-Debug-b11dca70f55dc20d4455.json","target-NightlyCoverage-Debug-ce4b96944ffe5b035829.json","target-NightlyMemCheck-Debug-c5e1e5fe032c03433c5e.json","target-NightlyMemoryCheck-Debug-7d77d6cd5d22eb2e2974.json","target-NightlyStart-Debug-5fdc4b6d0e4b07142b2c.json","target-NightlySubmit-Debug-3289d02ff4ab3279210d.json","target-NightlyTest-Debug-b16778a2eb1a15bd02ed.json","target-NightlyUpdate-Debug-89959c7a76a9003a42fc.json","target-docs-xxxxxxxxxx-html_zip-Debug-bba875f687bb0116730a.json","target-doxydoc-Debug-cb0e1dd8198d3cb35b9a.json","target-doxygen_doc-Debug-c43e59f5560a2cf6a3ec.json","target-laserscanner-Debug-3591bf9ce61cf09b363b.json","target-ndc8_parameter-Debug-6e53908ec7feeca701eb.json","target-test_bb_handler-Debug-9f74d551ebf92cff671d.json","target-test_laserscanner-Debug-8a64c4684efe5f696f94.json","toolchains-v1-2781238ec0fa0b9996f9.json"]
[extension] Not updating the configuration provider because "C_Cpp.intelliSenseEngine" is set to "Disabled"
[build] Build finished with exit code 0
[main] Copying compile_commands.json from /workspaces/xxxxxxx/build/native/compile_commands.json to /workspaces/xxxxxxx/compile_commands.json
[main] Building folder: xxxxxxx 
[main] Saving open files before configure/build
[build] Starting build
[driver] Start build all
[rollbar] Unhandled exception: Unhandled Promise rejection: cleanRebuild TypeError: inputs is not iterable {}
```

**Comment by benmcmorran** [comment]
Thanks for the report. I think this issue may have been introduced by #3009 (and is triggered by a missing `targets` property in `tasks.json`), but I'm having trouble reproducing the issue to be sure. To help narrow it down, can you provide the `tasks.json` that you're using (if any) and tell me how you are starting your build (e.g. running a task, invoking a certain command from the command palette, clicking a button in the statusbar, etc.)? Also if there's anything unusual about your CMake project please let me know.

**Comment by AndersHogqvist** [comment]
I'm happy to provide you with as much info as possible. I don't know what can be categorized as "unusual" but some general info about this project:

- I'm working in a dev container with Ubuntu
- It's a cross-compile project with two cmake kits targeting x64 and ARM
- I have a toolchain/build environment for ARM

Here are some related files (including tasks.json) used in the project. Just ask if there's anything else you need.

Here's the tasks.json:
```
{
  // See https://go.microsoft.com/fwlink/?LinkId=733558
  // for the documentation about the tasks.json format
  "version": "2.0.0",
  "tasks": [
    {
      "type": "cmake",
      "label": "build",
      "command": "build",
      "options": {
        "pwd": "build"
      },
      "problemMatcher": [
        {
          "owner": "cpplint",
          "fileLocation": [
            "absolute"
          ],
          "pattern": [
            {
              "regexp": "^(.*):\\s(.*)\\s(.*)$",
              "severity": 1,
              "code": 2
            },
            {
              "regexp": "^(.*):(\\d+):\\s\\s(.*)\\s\\s+\\[(.*)\\]$",
              "file": 1,
              "line": 2,
              "message": 3,
              // "": 4, Unknown what the last integer is indicating
              "loop": true
            }
          ]
        },
      ],
      "group": {
        "kind": "build",
        "isDefault": true
      },
    },
    {
      "type": "shell",
      "label": "clean-debug",
      "command": "cmake",
      "args": [
        "--build",
        "${workspaceFolder}/build/Debug",
        "--target",
        "clean"
      ],
      "problemMatcher": [],
    },
    {
      "type": "shell",
      "label": "clean-release",
      "command": "cmake",
      "args": [
        "--build",
        "${workspaceFolder}/build/Release",
        "--target",
        "clean"
      ],
      "problemMatcher": [],
    },
    {
      "label": "upload to xxx",
      "type": "shell",
      "command": "xxx 100.100.100.${input:device} upload ${input:file} --restart"
    }
  ],
  "inputs": [
    {
      "id": "device",
      "description": "choose device",
      "default": "161",
      "type": "pickString",
      "options": [
        "121",
        "131",
        "151",
        "161",
        "171",
        "203"
      ],
    },
    {
      "id": "file",
      "description": "choose file location",
      "default": "build/Debug/xxxx.zip",
      "type": "promptString",
    }
  ]
}
```

cmake-kits.json:
```
[
    {
        "name": "uzone",
        "toolchainFile": "${workspaceFolder}/cmake/xxxx-toolchain.cmake",
        "environmentVariables": {
            "OECORE_SDK_VERSION": "P41988V3.5.0"
        }
    },
    {
        "name": "native",
        "compilers": {
            "C": "/usr/bin/gcc",
            "CXX": "/usr/bin/g++"
        },
        "environmentVariables": {
            "BUILD_DIR": "build"
        }
    }
]
```

cmake-variants.yaml:
```
buildType:
  default: debug
  choices:
    debug:
      short: Debug
      buildType: Debug
    release:
      short: Release
      buildType: Release

buildDocs:
  default: 'no'
  choices:
    'yes':
      short: BuildDocs
      long: Build code documentation
      settings:
        XXXXXXXXX_ENABLE_DOCS: YES
    'no':
      short: NoDocs
      long: Don't build documentation
      settings:
        XXXXXXXXX_ENABLE_DOCS: NO
```

toolchain-file:
```
set(CMAKE_SYSTEM_NAME Linux)
set(CMAKE_SYSTEM_PROCESSOR ARM)

set(TOOLCHAIN_PREFIX arm-ndclinux-linux-gnueabi-)

set(ARM_TOOLCHAIN_DIR /opt/xxx/xxx/P41988V3.5.0/sysroots/x86_64-ndclinuxsdk-linux/usr/bin/arm-ndclinux-linux-gnueabi)

set(CMAKE_C_COMPILER ${ARM_TOOLCHAIN_DIR}/${TOOLCHAIN_PREFIX}gcc CACHE PATH "c-compiler")
set(CMAKE_ASM_COMPILER ${ARM_TOOLCHAIN_DIR}/${CMAKE_C_COMPILER} CACHE PATH "asm-compiler")
set(CMAKE_CXX_COMPILER ${ARM_TOOLCHAIN_DIR}/${TOOLCHAIN_PREFIX}g++ CACHE PATH "cxx-compiler")

set(CMAKE_C_FLAGS_INIT "-Wno-psabi -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9")
set(CMAKE_CXX_FLAGS_INIT "-Wno-psabi -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9")
set(CMAKE_ASM_FLAGS_INIT "")
set(CMAKE_EXE_LINKER_FLAGS_INIT "")

set(CMAKE_OBJCOPY ${ARM_TOOLCHAIN_DIR}/${TOOLCHAIN_PREFIX}objcopy CACHE INTERNAL "objcopy tool")
set(CMAKE_SIZE_UTIL ${ARM_TOOLCHAIN_DIR}/${TOOLCHAIN_PREFIX}size CACHE INTERNAL "size tool")

set(CMAKE_SYSROOT /opt/ndc/xxxxx/P41988V3.5.0/sysroots/cortexa9t2hf-neon-ndclinux-linux-gnueabi)
set( CMAKE_FIND_ROOT_PATH ${CMAKE_SYSROOT} )
set( CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER )
set( CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH )
set( CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH )
set( CMAKE_FIND_ROOT_PATH_MODE_PACKAGE BOTH )
```

**Comment by AndersHogqvist** [comment]
Adding `"targets": ["all"]` to the cmake build task in tasks.json solved it! Now I can build with v.1.14.30

**Comment by MikeTuev** [comment]
> Adding `"targets": ["all"]` to the cmake build task in tasks.json solved it! Now I can build with v.1.14.30

Thanks. it works for me to.
But now I'm getting the following on the console so every time it rebuilds the project, it first doing clean.

```
The defined targets in this task are being ignored.
clean task started....
/usr/local/bin/cmake --build /project/build --config Debug --target clean -j 8 --
clean finished successfully.
build task started....
/usr/local/bin/cmake --build /project/build --config Debug --target my_target -j 8 --
```
 
Do you expecting the same?

**Comment by AndersHogqvist** [comment]
> But now I'm getting the following on the console so every time it rebuilds the project, it first doing clean.

I don't have that issue.

**Comment by MikeTuev** [comment]
> I don't have that issue. Are you using the command palette and "CMake: Build" to start the build?
Yes. It's also binded to F7 by default.

I've tried to downgrade to 1.13.45 and the following popup now appears (with "targets": ["all"] added to tasks.json) :
![image](https://user-images.githubusercontent.com/25384351/233977033-fe5305bf-b81a-43dc-aca0-564f357c7ac0.png)

my tasks.json now looks like this:

```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Build the active target",
            "type": "cmake",
            "command": "build",
            "targets" : ["all"],
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

**Comment by benmcmorran** [comment]
I've tracked this down to a combination of `cmake.buildTasks` being `true` while `targets` is not set in `tasks.json` and have PR out with a fix. @MikeTuev I believe that popup is expected after you add `"targets": ["all"]` because the extension is now having trouble figuring out which task is a best fit for your currently selected build targets. The popup should go away after we get the fix out and you revert your `tasks.json` changes.

**Comment by benmcmorran** [comment]
This fix is included in version 1.14.31 that was just released to the marketplace.
